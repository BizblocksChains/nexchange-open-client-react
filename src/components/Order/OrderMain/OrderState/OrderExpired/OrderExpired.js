import React, { Component } from 'react';
import OrderAlert from '../OrderIcons/OrderAlert/OrderAlert';
import { I18n, Trans } from 'react-i18next';
import styles from '../OrderState.scss';

class OrderExpired extends Component {
  componentDidMount() {
    $('[data-toggle="tooltip"], [rel="tooltip"]').tooltip();
  }

  render() {
    return (
      <I18n ns="translations">
        {t => (
          <div className={styles.container}>
            <OrderAlert />
            <h2 className={styles.title}>{t('order.expired1')}</h2>

            <Trans i18nKey="order.expired2">
              <h3 className={styles.subtitle}>
                Your order is expired. You can make a{' '}
                <a href="/" className={styles['new-order-link']}>
                  new exchange order
                </a>.
              </h3>
            </Trans>

            <h4 className={styles.warning} data-toggle="tooltip" data-placement="top" data-original-title={t('order.expired3')}>
              {t('order.expired4')}
            </h4>
          </div>
        )}
      </I18n>
    );
  }
}

export default OrderExpired;
