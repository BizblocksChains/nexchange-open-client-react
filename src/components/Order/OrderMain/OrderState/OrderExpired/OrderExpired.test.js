import React from 'react';
import { shallow } from 'enzyme';
import OrderExpired from './OrderExpired';
window.$ = window.jQuery = require('jquery');
require('Js/bootstrap.min.js');

describe('OrderExpired', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<OrderExpired />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
