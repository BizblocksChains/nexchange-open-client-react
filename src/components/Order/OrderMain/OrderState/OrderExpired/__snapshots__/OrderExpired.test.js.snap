// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderExpired renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <OrderAlert />,
        <h2
          className="title"
>
          order.expired1
</h2>,
        <Trans
          i18nKey="order.expired2"
>
          <h3
                    className="subtitle"
          >
                    Your order is expired. You can make a
                     
                    <a
                              className="new-order-link"
                              href="/"
                    >
                              new exchange order
                    </a>
                    .
          </h3>
</Trans>,
        <h4
          className="warning"
          data-original-title="order.expired3"
          data-placement="top"
          data-toggle="tooltip"
>
          order.expired4
</h4>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "order.expired1",
          "className": "title",
        },
        "ref": null,
        "rendered": "order.expired1",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h3
            className="subtitle"
>
            Your order is expired. You can make a
             
            <a
                        className="new-order-link"
                        href="/"
            >
                        new exchange order
            </a>
            .
</h3>,
          "i18nKey": "order.expired2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Your order is expired. You can make a",
              " ",
              <a
                className="new-order-link"
                href="/"
>
                new exchange order
</a>,
              ".",
            ],
            "className": "subtitle",
          },
          "ref": null,
          "rendered": Array [
            "Your order is expired. You can make a",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "new exchange order",
                "className": "new-order-link",
                "href": "/",
              },
              "ref": null,
              "rendered": "new exchange order",
              "type": "a",
            },
            ".",
          ],
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "order.expired4",
          "className": "warning",
          "data-original-title": "order.expired3",
          "data-placement": "top",
          "data-toggle": "tooltip",
        },
        "ref": null,
        "rendered": "order.expired4",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <OrderAlert />,
          <h2
            className="title"
>
            order.expired1
</h2>,
          <Trans
            i18nKey="order.expired2"
>
            <h3
                        className="subtitle"
            >
                        Your order is expired. You can make a
                         
                        <a
                                    className="new-order-link"
                                    href="/"
                        >
                                    new exchange order
                        </a>
                        .
            </h3>
</Trans>,
          <h4
            className="warning"
            data-original-title="order.expired3"
            data-placement="top"
            data-toggle="tooltip"
>
            order.expired4
</h4>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "order.expired1",
            "className": "title",
          },
          "ref": null,
          "rendered": "order.expired1",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h3
              className="subtitle"
>
              Your order is expired. You can make a
               
              <a
                            className="new-order-link"
                            href="/"
              >
                            new exchange order
              </a>
              .
</h3>,
            "i18nKey": "order.expired2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Your order is expired. You can make a",
                " ",
                <a
                  className="new-order-link"
                  href="/"
>
                  new exchange order
</a>,
                ".",
              ],
              "className": "subtitle",
            },
            "ref": null,
            "rendered": Array [
              "Your order is expired. You can make a",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "new exchange order",
                  "className": "new-order-link",
                  "href": "/",
                },
                "ref": null,
                "rendered": "new exchange order",
                "type": "a",
              },
              ".",
            ],
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "order.expired4",
            "className": "warning",
            "data-original-title": "order.expired3",
            "data-placement": "top",
            "data-toggle": "tooltip",
          },
          "ref": null,
          "rendered": "order.expired4",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
